doc/htmldoc/neurons/model_details/IAF_Integration_Singularity.ipynb

```diff
@@ -0,0 +1,945 @@
+      "text/plain": [
+       "<Figure size 1200x400 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "tau_m = 10.\n",
+    "tau_s = tau_m + np.logspace(-12, 0, 26)\n",
+    "h = np.array([[1, 0.1, 0.01, 0.001]]).T\n",
+    "p31, p31s, p32, p32s = calc_p(tau_m, tau_s, h)\n",
+    "\n",
+    "fig = plt.figure(figsize=(12, 4));\n",
+    "ax1 = fig.add_subplot(1, 2, 1);\n",
+    "ax1.loglog(np.abs(tau_s-tau_m), np.abs(p31-p31s).T, 'o-');\n",
+    "ax1.set_ylabel('$P - P_{singular}$')\n",
+    "ax1.set_ylim([1e-20, 5e-2])\n",
+    "ax1.set_xlabel(r'$\\tau_s - \\tau_m$ [ms]');\n",
+    "ax1.set_title('$P_{31}$ difference from singular value');\n",
+    "ax2 = fig.add_subplot(1, 2, 2);\n",
+    "ax2.loglog(np.abs(tau_s-tau_m), np.abs(p32-p32s).T, 'o-', \n",
+    "           label=[f'h = {hv:.3f} ms' for hv in h[:, 0]]);\n",
+    "ax2.set_ylim([1e-20, 5e-2]);\n",
+    "ax2.set_xlabel(r'$\\tau_s - \\tau_m$ [ms]');\n",
+    "ax2.set_title('$P_{32}$ difference from singular value');\n",
+    "plt.legend();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- $P_{32}$ shows perfect convergence towards the singular value up to the limits of numerical accuracy\n",
+    "    - This holds for all step sizes $h$\n",
+    "    - This is plausible because $P_{32}$ contains multiplications only: `gamma * np.exp(-h/tau_s) * np.expm1(h*inv_beta)`\n",
```

Comment:
Not sure I follow plausibility argument; as far as I see it's actually the use of ``expm1()`` that saves us here (using ``exp() - 1`` exhibits similar issues as $P_{31}$).

Reply:
Yes, that is exactly my point: by using `expm1()` (which we also do in the C++ implementation in NEST), we end up with an expression for `P_32` which contains no differences at the C++ level and thus has no risk of introducing significant numerical noise by subtracting two almost identical numbers from each other. For `P_31` on the other hand, we have and explicit difference that we cannot avoid.